core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java

```diff
@@ -646,6 +659,15 @@ private DefaultFunctionResolver date_format() {
     );
   }
 
+  private ExprValue dayOfYearToday(Clock clock) {
+    return new ExprIntegerValue((formatNow(clock).getDayOfYear()));
```

Comment:
You can avoid extra actions made in `formatNow`
```suggestion
    return new ExprIntegerValue(LocalDateTime.now(clock).getDayOfYear());
```

Reply:
Fixed in 2517b31f4ed5688dcaa7baa78aa4a025a71a28a4